test/tbb/test_join_node_preview.cpp

```diff
@@ -0,0 +1,98 @@
+/*
```

Comment:
I see this file is supposed to test preview features of the join node. Yet, it seems to test a regression (not necessarily connected with preview features such as `follows` and `precedes`). Also, I see message key matching is placed in separate test files, though it is also a preview feature. I wonder how can we organize tests for join node better?

Reply:
I admit the possibility that I overdid the splitting because only splitting message key matching test was really helpful to solve the issue and maybe this `test_join_node_preview` is not needed...

Reply:
As for `-DTBB_TEST_LOW_WORKLOAD` approach (I can't reply to thread thread below) I guess we just reduce scope of testing on MIPS but the problem is not high workload but a high number of symbols. Splitting testing into separate file doesn't affect test scope at all.

Reply:
What if we move all the test of preview features of the join_node inside this file. Will that work?

Reply:
This file already contains code parts under PREVIEW macro from other *join_node* testing files. Exception is `test_join_node_msg_key_matching`.

Reply:
As per offline discussion, moving all the tests of preview features inside this single file will presumably break linkage again.